### PR TITLE
Update Azurite and Azurite tests to ref t2 Azure Storage SDK

### PR DESCRIPTION
**Pull request by zzhxiaofeng**
Azurite is upgraded to 3.8.0 now, so I closed #455 , and open this PR for this change. 
Change summary:
1. Update Azurite to ref t2 Azure Storage SDK.
2. Update Azurite tests to ref t2 Azure Storage SDK.

We noticed there are ci errors for some checks, but many checks passed. We found this is because the t2 Azure Storage SDK not robust while doing some operation like delete services.

Error:
Annotation 2020-06-17 103855

We filed a issue in Azure SDK repo: https://github.com/Azure/azure-sdk-for-js/issues/10065

### CODE REVIEW

**Comment by blueww** [comment]
@zzhxiaofeng  
Please help to look at the CI failure.

**Comment by blueww** [comment]
@zzhxiaofeng
Thanks for raising the PR!

Beside SDK upgrade and the common test case change (API update from Track1 to Track2), would you please summarize what's the other change you have done (like if any wait time add in test case), it can help for a more efficient PR review.

Thanks!

**Comment by zzhxiaofeng** [comment]
I file a https://github.com/Azure/azure-sdk-for-js/issues/10065 in Azure SDK to check this.

**Comment by zzhxiaofeng** [comment]
@blueww  I have updated PR description. Please have a look if there are any other questions.

**Comment by blueww** [comment]
@zzhxiaofeng 
Thanks for the update!

Please make sure the CI can pass.

**Comment by XiaoningLiu** [comment]
Re-run test cases

**Comment by XiaoningLiu** [comment]
@zzhxiaofeng One test case is failing, interesting why socket will hang, does your local run scuessful?

  1) BlockBlobHighlevel
       "after each" hook for "uploadFile should success when blob >= BLOCK_BLOB_MAX_UPLOAD_BLOB_BYTES @loki @sql":
     FetchError: request to http://127.0.0.1:11000/devstoreaccount1/container159522884259403144?restype=container failed, reason: socket hang up
      at ClientRequest.<anonymous> (node_modules\node-fetch\lib\index.js:1455:11)
      at Socket.socketOnEnd (_http_client.js:437:9)
      at endReadableNT (_stream_readable.js:1064:12)
      at _combinedTickCallback (internal/process/next_tick.js:139:11)
      at process._tickCallback (internal/process/next_tick.js:181:9)

**Comment by zzhxiaofeng** [comment]
All tests passed while running in local. These failures just show in CI. Can you help?

**Comment by blueww** [comment]
@zzhxiaofeng 
As we see, the failure all happen in group "BlockBlobHighlevel", but can happen on different cases in the group.
It seems the failure happens random, and might be related with the before cases running.

Would you like try to run the full Unit test several time, and see if this can repro on your local machine?

**Comment by zzhxiaofeng** [comment]
In my local machine, would fail like 1 time in 10.

**Comment by blueww** [comment]
@zzhxiaofeng 

As you can repro it on your local machine, would like see how to fix it in Unit test?

**Comment by zzhxiaofeng** [comment]
@blueww Finally, all checks passed. Please review.

**Comment by blueww** [comment]
@zzhxiaofeng 
Thanks for fix the CI!

@XiaoningLiu 
I have go through the PR, and generally good with it.
Please help to review https://github.com/Azure/Azurite/pull/472/commits/1a93d6a147c18547dc22bf81d710945624ffff7c, and see if you are OK with the change?
Besides that, do we need to run CI more times to make sure it's stable enough?

**Comment by blueww** [comment]
apz /rerun

**Comment by jongio** [comment]
We're still seeing CI issues.  @zzhxiaofeng Can you please help investigate?

File: tests/blob/blockblob.highlevel.test.ts [hunk 1]
```diff
@@ -36,38 +31,56 @@ describe("BlockBlobHighlevel", () => {
...
   });
 
-  afterEach(async () => {
-    await containerURL.delete(Aborter.none);
+  afterEach(async function () {
+    const containerUrl = containerClient.url;
+    try {
+      await containerClient.delete();
+    } catch (error) {
+      this.timeout(3000);
```
Comment (blueww) [inline tests/blob/blockblob.highlevel.test.ts:73]: @XiaoningLiu 
Please help to look.

**Comment by XiaoningLiu** [comment]
> /azp run

Sent you a GitHub invitation to join Azurite repo, can you accept and see if you can goahead with `/azp run`?

**Comment by XiaoningLiu** [comment]
Synced with @zzhxiaofeng offline, should be an socket issue about track2 SDK keep alive option. Track2 SDK enables keep alive by default, turn it off in test cases may help. While we should follow up with socket issue in another issue discussion.

**Pull request merged**